### PR TITLE
TimeRangePicker: Announce to screen reader when time range is updated

### LINKED ISSUE #86834: [Drill]: Screen reader is staying silent, when 'Next' and 'Previous' buttons are activated.
### What happened?

Screen reader is not announcing the updated time zone when 'Next' and 'Previous' buttons are activated. 

Screen reader stays silent. 

### What did you expect to happen?

Screen reader should announce the updated time zone when 'Next' and 'Previous' buttons are activated. 

### Did this work before?

No

### How do we reproduce it?

1.Open the URL in edge browser [Grafana Play Home - Dashboards - Grafana](https://play.grafana.org/d/000000012/grafana-play-home?orgId=1). 

2.‘Grafana Play Home’ page will open. 

3.Navigate to ‘Time range’ button & expand it. 

4.Navigate to ‘From’ & ‘To’ fields, select respective dates from date picker dialog. 

5.Now, again navigate to ‘Forward’ & ‘backward’ buttons present for the time zone & activate them. 

6.Observe the screen reader narration. 

### Is the bug inside a dashboard panel?

_No response_

### Environment (with versions)?

Grafana: v11.1.0-69051 (9d44c8e8cf)
OS: Version 24H2 (OS Build 26100.1) 
Browser: Version 125.0.2506.0 (Official build) dev (64-bit) 

### Grafana platform?

None

### Datasource(s)?

_No response_

### PR DESCRIPTION
**Pull request by tskarhed**
**What is this feature?**

Creates an aria live region for the label on the time range picker.

**Why do we need this feature?**

So you can know when the time range is updated.

**Who is this feature for?**

Screen reader users.

**Which issue(s) does this PR fix?**:

Fixes [#86834](https://github.com/grafana/grafana/issues/86834)

**Special notes for your reviewer:**
Test by updating the time range somehow. Zoom out or moving with the arrows.

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: packages/grafana-ui/src/components/DateTimePickers/TimeRangePicker.tsx [hunk 1]
```diff
@@ -235,7 +235,7 @@ export const TimePickerButtonLabel = memo<LabelProps>(({ hideText, value, timeZo
   }
 
   return (
-    <span className={styles.container}>
+    <span className={styles.container} aria-live="polite" aria-atomic="true">
```
Comment (tskarhed) [inline packages/grafana-ui/src/components/DateTimePickers/TimeRangePicker.tsx:238]: I feel `assertive` should be better here, as this update is usually an immediate consequence of a button press (or whatever other method you use to update it). It feels a bit sluggish with the delay, and might be harder to associate with the key press.

Comment (eledobleefe) [reply packages/grafana-ui/src/components/DateTimePickers/TimeRangePicker.tsx:238]: 🤔 According to the following, I'd go for `aria-live="polite"` as I don't think an update on this is critical 🤷‍♀️ 

> `aria-live="assertive"` should only be used for time-sensitive/critical notifications that absolutely require the user's immediate attention.

Comment (tskarhed) [reply packages/grafana-ui/src/components/DateTimePickers/TimeRangePicker.tsx:238]: But it is time sensitive, because it is generated by the user action.

Comment (tskarhed) [reply packages/grafana-ui/src/components/DateTimePickers/TimeRangePicker.tsx:238]: I think polite works fine with previous and next arrows in this case, but with the zoom out button it is a bit annoying.

**Review by ashharrison90 — APPROVED** [review]
imo i think this works great in all cases 👍 

great job 🥳

**Pull request merged**